File: ai-gateway/app/api/routes/health.py
"""健康检查接口。

前缀: ``/health``（无版本前缀）

用于负载均衡、容器探活；无需鉴权。
"""

from __future__ import annotations

from fastapi import APIRouter, Depends

from app.api.deps import get_state
from app.state import AppState

router = APIRouter(tags=["health"])


@router.get("/health")
async def health(state: AppState = Depends(get_state)) -> dict:
    """服务存活探测。

    - 生产环境仅返回 ``{"status": "ok"}``
    - 非生产环境额外返回已加载模型数、客户端数
    """
    if state.settings.app_env == "production":
        return {"status": "ok"}
    return {
        "status": "ok",
        "models_loaded": len(state.models.routes),
        "clients_loaded": state.clients.client_count,
    }

File: ai-gateway-go/internal/handler/health.go
package handler

import (
	"net/http"
	"strconv"

	"github.com/gin-gonic/gin"
	"github.com/porsche/ai-gateway-go/internal/app"
	"github.com/porsche/ai-gateway-go/internal/middleware"
)

func RegisterHealth(r *gin.Engine, state *app.State) {
	r.GET("/health", func(c *gin.Context) {
		if state.Settings.AppEnv == "production" {
			c.JSON(http.StatusOK, gin.H{"status": "ok"})
			return
		}
		c.JSON(http.StatusOK, gin.H{
			"status":          "ok",
			"models_loaded":   state.Models.Count(),
			"clients_loaded":  state.Clients.ClientCount(),
		})
	})
}

func RegisterAdmin(r *gin.Engine, state *app.State) {
	g := r.Group("/admin", middleware.RequireAdmin(state))
	g.GET("/status", func(c *gin.Context) {
		routes := map[string]string{}
		for name, route := range state.Models.Routes() {
			routes[name] = route.Provider
		}
		c.JSON(http.StatusOK, gin.H{
			"models":  state.Models.Count(),
			"clients": state.Clients.ClientCount(),
			"routes":  routes,
		})
	})
	g.POST("/reload-config", func(c *gin.Context) {
		state.ReloadConfig()
		c.JSON(http.StatusOK, gin.H{
			"status":  "reloaded",
			"models":  state.Models.Count(),
			"clients": state.Clients.ClientCount(),
		})
	})
}

func parseUintQuery(c *gin.Context, key string, def int) int {
	v := c.DefaultQuery(key, strconv.Itoa(def))
	n, err := strconv.Atoi(v)
	if err != nil {
		return def
	}
	return n
}
